File: Selenium/Activities/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxServices
from selenium.webdriver.remote import webelement
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxServices(GeckoDriverManager().install())
with webdriver.Firefox(service = service) as driver:
    wait = WebDriverWait(driver,10)
    driver.get("https://v1.training-support.net/selenium/dynamic-controls")
    print("Title of the page is :",driver.title)
    checkbox = driver.find_element(By.XPATH,"//div[@id ='dynamicCheckbox']//input")
    checkbox.click()
    print(checkbox.is_selected())
    driver.find_element(By.ID,"toggleCheckbox").click()
    print(wait.until(expected_conditions.invisibility_of_element_located(checkbox)))
    driver.find_element(By.XPATH,"//button[text() ='Add Checkbox']").click()
    wait.until(expected_conditions.element_located_to_be_selected((By.XPATH,"//div[@id ='dynamicCheckbox']//input")))
    print(checkbox.is_selected())
    driver.close()

File: Selenium/Activities/Activity10.java
package activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity10 {
    public static void main(String[] args){
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://v1.training-support.net/selenium/dynamic-controls");
        System.out.println("Title of the Page :" + driver.getTitle());
        WebElement checkbox = driver.findElement(By.xpath("//div[@id ='dynamicCheckbox']//input"));
       if(checkbox.isEnabled()){
            checkbox.click();
        }
        System.out.println("is checkbox selected :" + checkbox.isSelected());
        driver.findElement(By.id("toggleCheckbox")).click();
        System.out.println("Check box Removed :");
        System.out.println(wait.until(ExpectedConditions.invisibilityOf(checkbox)));
        System.out.println("Is checkbox available :" +checkbox.isDisplayed());
        driver.findElement(By.xpath("//button[text() ='Add Checkbox']")).click();
        wait.until(ExpectedConditions.visibilityOf(checkbox)).click();
        System.out.println("Is check box visible and selected :" + checkbox.isSelected());
        driver.close();
    }
}
